File: ungs-docsys-bff/src/modules/resume-user/controllers/resume-user.controller.ts
import { Controller, Post, Get, Param, Body, Headers, Query } from '@nestjs/common';
import { ApiOperation, ApiResponse } from '@nestjs/swagger';
import { ResumeUserService } from '../services/resume-user.service';
import { ResumeUserRequestDto } from '../dtos/resume-user-request.dto';
import { ResumeUserResponseDto } from '../dtos/resume-user-response.dto';

@Controller('/v1/resume-user')
export class ResumeUserController {
  constructor(private readonly resumeUserService: ResumeUserService) {}

  @Post()
  @ApiOperation({ summary: 'Save Resume User' })
  @ApiResponse({
    status: 200,
    description: 'Successfully saved resume user',
    type: ResumeUserResponseDto,
  })
  async saveResumeUser(
    @Headers('authorization') authorization: string,
    @Body() request: ResumeUserRequestDto,
  ): Promise<ResumeUserResponseDto> {
    return await this.resumeUserService.save(request, authorization);
  }

  @Get(':id')
  @ApiOperation({ summary: 'Get Resume user by id'})
  @ApiResponse({
      status: 200,
      description: 'Resume user by id',
      type: ResumeUserResponseDto,
    })
  getById(@Headers('authorization') authorization: string, @Param('id') id: number): Promise<ResumeUserResponseDto> {
    return this.resumeUserService.getById(id, authorization);
  }

  @Get()
  @ApiOperation({ summary: 'Get Resume user by id'})
  @ApiResponse({
      status: 200,
      description: 'Resume user by id',
      type: [ResumeUserResponseDto],
    })
  getByParams(@Headers('authorization') authorization: string, @Query('isCurrent') isCurrent: boolean): Promise<ResumeUserResponseDto[]> {
    return this.resumeUserService.getByParams(isCurrent, authorization);
  }
}

File: ungs-docsys-ms/src/main/java/com/ungs/docsys/controllers/ResumeUserController.java
package com.ungs.docsys.controllers;

import com.ungs.docsys.dtos.AppUserClaimDto;
import com.ungs.docsys.dtos.ResumeUserRequestDto;
import com.ungs.docsys.dtos.ResumeUserResponseDto;
import com.ungs.docsys.security.JwtUtil;
import com.ungs.docsys.services.ResumeUserService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/v1/resume-user")
@Validated
@AllArgsConstructor
public class ResumeUserController {

    private ResumeUserService resumeUserService;
    private final JwtUtil jwtUtil;

    @Operation(summary = "Create a resume for the user")
    @ApiResponse(responseCode = "200", description = "Resume created successfully")
    @PostMapping()
    public ResponseEntity<ResumeUserResponseDto> save(
            @RequestHeader("Authorization") String authorization,
            @RequestBody @Valid ResumeUserRequestDto resumeUserRequestDto) {
        return ResponseEntity.ok(resumeUserService.save(resumeUserRequestDto, jwtUtil.extractUserClaim(authorization)));
    }

    @Operation(summary = "Get resume by user ID")
    @ApiResponse(responseCode = "200", description = "Resume retrieved successfully")
    @GetMapping("/{id}")
    public ResponseEntity<ResumeUserResponseDto> getResumeByUserId(
            @PathVariable Long id) {
        return ResponseEntity.ok(resumeUserService.getById(id));
    }
}